tests/unit_tests/data_steward/gcloud/gcs/gcs_test.py

```diff
@@ -19,31 +28,42 @@ def setUpClass(cls):
         print('**************************************************************')
 
     def setUp(self):
-        self.client = StorageClient.create_anonymous_client()
```

Comment:
Why did you switch away from using the anonymous client constructor?

Reply:
It became apparent that this was not really addressing a critical point of pain when designing tests:  Auth happens on object creation.

So whenever a StorageClient is created, the upstream google client is initialized and an auth happens during this phase.  When creating an anonymous client, that auth is sort of "delayed" until an auth is determined to be appropriate.  When Nishanth and I were examining the tests, we found using google's create_anonymous_client() was causing errors with "auth tokens that couldn't be refreshed" or something to that effect between tests.  Mocking the auth process didn't work for us or for StackOverflow.

What we really needed was to create the client, but not call the upstream __init__ that forces the auth process, so the DummyClient() was created.   Using this dummy client, we bypass the __init__ call using a pass statement, and can then mock the upstream client's methods (list_blobs, etc.).

Reply:
thanks for the explanation :)